featuretools/tests/primitive_tests/test_primitive_utils.py

```diff
@@ -135,3 +138,94 @@ def test_errors_no_primitive_in_file(bad_primitives_files_dir):
+
+    for i in range(len(rolling_series_pd)):
+        start_idx = i - gap - window_length + 1
+        end_idx = i - gap
+
+        # If start and end are negative, they're entirely before
+        if start_idx < 0 and end_idx < 0:
+            assert pd.isnull(rolling_max.iloc[i])
+            assert pd.isnull(rolling_min.iloc[i])
+            continue
+
+        if start_idx < 0:
+            start_idx = 0
+
+        # Because the row values are a range from 0 to 20, the rolling min will be the start index
+        # and the rolling max will be the end idx
+        assert rolling_min.iloc[i] == start_idx
+        assert rolling_max.iloc[i] == end_idx
+
+
+def test_roll_series_with_no_gap(rolling_series_pd):
+    window_length = 3
+    gap = 0
+    actual_rolling = _roll_series_with_gap(rolling_series_pd, window_length, gap=gap).mean()
+    expected_rolling = rolling_series_pd.rolling(window_length, min_periods=1).mean()
+
+    pd.testing.assert_series_equal(to_pandas(actual_rolling), to_pandas(expected_rolling))
+
+
+@pytest.mark.parametrize(
+    "window_length, gap",
+    [
+        (6, 2),
+        (6, 0)  # No gap - changes early values
+    ]
+)
+def test_roll_series_with_gap_early_values(window_length, gap, rolling_series_pd):
+    # Default min periods is 1 - will include all
+    default_partial_values = to_pandas(_roll_series_with_gap(rolling_series_pd,
+                                                             window_length,
+                                                             gap=gap).count())
+    num_empty_aggregates = len(default_partial_values.loc[default_partial_values == 0])
+    num_partial_aggregates = len((default_partial_values
+                                  .loc[default_partial_values != 0])
+                                 .loc[default_partial_values < window_length])
+    assert num_empty_aggregates == gap
+    assert num_partial_aggregates == window_length - 1
+
+    # Make min periods the size of the window
+    no_partial_values = to_pandas(_roll_series_with_gap(rolling_series_pd,
+                                                        window_length,
+                                                        gap=gap,
+                                                        min_periods=window_length).count())
+    num_null_aggregates = len(no_partial_values.loc[pd.isna(no_partial_values)])
+    num_partial_aggregates = len(no_partial_values.loc[no_partial_values < window_length])
+
+    # because we shift, gap is included as nan values in the series.
+    # Count treats nans in a window as values that don't get counted,
+    # so the gap rows get included in the count for whether a window has "min periods".
+    # This is different than max, for example, which does not count nans in a window as values towards "min periods"
+    assert num_null_aggregates == window_length - 1
+    assert num_partial_aggregates == gap
+
+
+def test_roll_series_with_gap_nullable_types(rolling_series_pd):
```

Comment:
Any need to also test these with an input series that contains null values to confirm they are handled as expected?

Reply:
yep, good idea--will add